packaging/cmake/Modules/NetdataDetectSystemd.cmake

```diff
@@ -4,39 +4,43 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 macro(detect_systemd)
-    find_library(SYSTEMD_LIBRARY NAMES systemd)
+    if(LINUX)
```

Comment:
Don't add the OS logic within detect_systemd, instead call detect_systemd only on Linux.  No need to `set(HAVE_SYSTEMD False)` as this is the default for unset variables.